tic_tac_toe/board.py

```diff
@@ -29,8 +38,27 @@ def current_player(self):
         return "X" if self.turn_count() % 2 == 0 else "O"
 
     def position_taken(self, index):
-        return self.board[index] == "X" or self.board[index] == "O"
+        return self.board[index] != " "
 
     def within_bounds(self, index):
         return index >= 0 and index <= 8
 
+    def win(self):
+        won = False
+        for combination in self.win_combinations:
```

Comment:
You could try expanding the indices. I'm thinking it might improve the readability, but I'm curious what you think.

```
for i0, i1, i2 in in self.win_combinations:
    # ...
```